compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDmaLoopSubsumption.cpp

```diff
@@ -289,17 +289,36 @@ struct SubsumeLoopIntoDMA
                 insertSourceStrides, updateSourceOffsets))) {
           return failure();
         }
+
```

Comment:
What about a 

```
auto getAppended = [](ArrayRef<OpFoldResult> prefix, ArrayRef<int64_t> suffix) -> SmallVector<int64_t> {
  // ...
};
```

to reduce dup? Then

```
auto newSourceSizesInt = getAppended(newSourceSizes, insertSourceSizes); 
auto newSourcesStridesInt = getAppended(...);
...

Reply:
Very good point! I've made the change.